ionc/ion_allocation.c

```diff
@@ -224,7 +220,8 @@ void _ion_free_block(ION_ALLOCATION_CHAIN *pblock)
 void ion_initialize_page_pool(SIZE page_size, int free_page_limit)
 {
     // once the page list is in use, you can't change your mind
-    if (g_ion_alloc_page_list.page_size != ION_ALLOC_PAGE_POOL_PAGE_SIZE_NONE)
+    // This is changed, because g_ion_alloc_page_list.page_size is initialized to ION_ALLOC_PAGE_POOL_DEFAULT_PAGE_SIZE.
+    if (g_ion_alloc_page_list.page_size == ION_ALLOC_PAGE_POOL_DEFAULT_PAGE_SIZE)
```

Comment:
Won't this always be true (note: this is `==`, not `!=` as before)? `g_ion_alloc_page_list.page_size` is statically initialized to `ION_ALLOC_PAGE_POOL_DEFAULT_PAGE_SIZE` as part of this change. And I don't think the solution is just to make this `!=` because that doesn't allow us to distinguish between the default and a user-selected value that happens to be the same as the default.

Now, this probably isn't a big deal since `ion_alloc.h` isn't actually part of the public includes, so users can't really call `ion_initialize_page_pool` manually anyway. It seems like it would be desirable for this to be configurable, either directly by calling `ion_initialize_page_pool` or indirectly via `ION_WRITER_OPTIONS`. However, we don't need to solve that as part of this change.

I would at least like to have a comment in here that acknowledges that `ion_initialize_page_pool` does nothing so that I can create an issue to look into addressing that, though.